lib/galaxy/managers/datasets.py

```diff
@@ -338,7 +338,7 @@ def serialize_permissions(self, item, key, user=None, **context):
 
 # ============================================================================= AKA DatasetInstanceManager
 class DatasetAssociationManager(
-    base.ModelManager[model.DatasetInstance],
+    base.ModelManager[model.DatasetInstance],  # type:ignore[type-var]
```

Comment:
That doesn't look right either, what's the problem ?

Reply:
This all makes sense to me I think - the type bound is specified to be "Base" and DatasetInstance isn't a subset of Base class. Can we retain _HasTable for everything but also make the Base change for all the Declaratively mapped things and then kill _HasTable if we ever are able to declaratively map everything?